lib/streams/compound-duplex.js

```diff
@@ -102,4 +109,37 @@ CompoundDuplex.prototype._abortStream = function() {
 	}
 };
 
+CompoundDuplex.prototype.wrap = function(stream) {
```

Comment:
I don't like this kind of structure at all.  It opens up the possibility of things like constructing a RequestStream and then calling wrap(), which doesn't make any semantic sense.  As I mentioned earlier, it would make much more sense to either create a WrapperStream that extends CompoundDuplex (this would be ideal given that you're adding additional fields and methods), or to make a static wrap() method on the constructor.